### PR TITLE
feat: allow destructive changes with MetadataOptions

### PR DESCRIPTION
**Pull request by shetzel**
### What does this PR do?
Allows defining metadata type/name pairs to be deleted (pre or post) as part of a metadata deploy.  Adds snapshot tests for this behavior.

### What issues does this PR fix or reference?
@W-15306474@

### CODE REVIEW

File: test/snapshot/sampleProjects/destructiveCompSets/snapshots.test.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    });
+
+    const convertedFiles = await getConvertedFilePaths(postTestOutputDir);
+    for (const file of convertedFiles) {
+      await fileSnap(file, testDir);
+    }
+    dirsAreIdentical(path.join(snapshotsDir, 'testOutput', 'post1'), postTestOutputDir);
+  });
+
+  it('2 post-destructive and 2 deployments', async () => {
```
Comment (WillieRuemmele) [inline test/snapshot/sampleProjects/destructiveCompSets/snapshots.test.ts:113]: should/could we add a test covering 1+ pre or post destructive changes, without any constructive? Seems like it'll be working - but would be good to have for future changes

Comment (WillieRuemmele) [reply test/snapshot/sampleProjects/destructiveCompSets/snapshots.test.ts:113]: or maybe we don't want to support that?  if the CLIs not adding new flags to use this, I'm not sure if other SDR consumers would like that ability

**Review by WillieRuemmele — APPROVED** [review]

File: test/snapshot/sampleProjects/destructiveCompSets/snapshots.test.ts (hunk 1 above)
Comment (shetzel) [reply test/snapshot/sampleProjects/destructiveCompSets/snapshots.test.ts:113]: I can add a test for those cases.  👍

**Comment by WillieRuemmele** [comment]
### QA Notes
> since there's no flags, or exposure through the CLI, I built a small script to test the behavior

```ts
    const c = await ComponentSetBuilder.build({metadata:{
      directoryPaths:['force-app/main/default/classes'],
        destructiveEntriesPost: ['ApexClass:myClassPOST'],
        destructiveEntriesPre: ['ApexClass:myClassPRE'],
        metadataEntries: ['ApexClass:myClass']
      }})

    const p = await c.getPackageXml(4, DestructiveChangesType.PRE);
    const x = await c.getPackageXml(4, DestructiveChangesType.POST);

    // eslint-disable-next-line no-console
    console.log(p,x);
```

✅ : I ensured that both variables, `p`/`x` are correct.
✅ : when deployed, the classes were deleted

**Pull request merged**